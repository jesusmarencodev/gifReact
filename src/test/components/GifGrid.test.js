import React from 'react';
import '@testing-library/jest-dom'; //autocompletado mejorado
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas componente <GifGrid/>', () => {
	let category = 'Naruto';
	


	test('Debe de mostrar correctamente el componente ', () => {

/* 																Nota importante 
 					Cuando se va a hacer un mock el wrapper debe ir despues del mock
					porque sino va a fallar la prueba		 
*/	 			
		
		/* Mockeando data en el custom hook useFetchGifs */
		useFetchGifs.mockReturnValue({ //estado inicial del mock
			data : [],
			loading : true
		})

		const wrapper = shallow(<GifGrid category={category} />);

		expect(wrapper).toMatchSnapshot();
	});

	test('Debe de mostrar items cuando se cargan imagenes en el useFetchGifs', () => {
		const gifs = [{
				id : 'ABC',
				url : 'https://localhost/cualquier/cosa.jpg',
				title : 'cualquier cosa'
		}]

		/* Mockeando data en el custom hook useFetchGifs  que ya trae data*/
		useFetchGifs.mockReturnValue({ //estado inicial del mock
			data : gifs,
			loading : false
		})

		/* Crear el wrapper */
		const wrapper = shallow(<GifGrid category={category} />);

		//genera un nuevo snapshop, no sobreescribe, lo pinta debajo del otro
		expect(wrapper).toMatchSnapshot();

		//si viene data no deberia existir el parrafo con loading
		expect(wrapper.find('p').exists()).toBe(false);

		//si viene data en el useFetchGifs deberia de existir
		// el mismo numero de componentes GifGridItem, en este caso el 
		// .length del mock gifs
		expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

	})

	test('Debe mostrar en el h3 el value que llega en categoria', () => {
		const wrapper = shallow(<GifGrid category={category} />);

		expect(wrapper.find('h3').text()).toBe(category);
	})
	
	
	

});
