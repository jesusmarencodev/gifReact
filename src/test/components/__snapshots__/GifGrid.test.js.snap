// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas componente <GifGrid/> Debe de mostrar correctamente el componente  1`] = `
<Fragment>
  <h3>
    Naruto
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas componente <GifGrid/> Debe de mostrar items cuando se cargan imagenes en el useFetchGifs 1`] = `
<Fragment>
  <h3>
    Naruto
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;
